pkg/pillar/types/zedroutertypes.go

```diff
@@ -128,6 +128,50 @@ func (status AppNetworkStatus) VerifyFilename(fileName string) bool {
 			fileName, expect)
 	}
 	return ret
+
+}
+
+// PerInterfaceErrorMap - Used to return per-interface errors.
+//  ifName is usually used as the key
+type PerInterfaceErrorMap struct {
+	IntfErrorMap map[string]ErrorInfo
+}
+
+// AddError - Add an error into PerInterfaceErrorMap. If en entry already
+//  exists for the interface, it is appended
+func (intfMap *PerInterfaceErrorMap) AddError(ifName string,
+	err error, source string) {
+	errInfo, ok := intfMap.IntfErrorMap[ifName]
+	if ok {
+		errInfo.Error = errInfo.Error + " \n " + err.Error()
+		if errInfo.ErrorSource == "" {
+			errInfo.ErrorSource = source
+		} else {
+			errInfo.ErrorSource += ", " + source
+		}
+	} else {
+		errInfo = NewErrorInfo(err, source)
+	}
+	intfMap.IntfErrorMap[ifName] = errInfo
+}
+
+// DelError - Delete inteface entry
+func (intfMap *PerInterfaceErrorMap) DelError(ifName string) {
+	delete(intfMap.IntfErrorMap, ifName)
+}
+
+// AddMap - Add all the entries from the given per-interface map
+func (intfMap *PerInterfaceErrorMap) AddMap(target PerInterfaceErrorMap) {
+	for intf, errInfo := range target.IntfErrorMap {
+		intfMap.AddError(intf, errors.New(errInfo.Error), errInfo.ErrorSource)
+	}
+}
+
+// NewPerInterfaceErrorMap - Create a new instance of PerInterfaceErrorMap
+func NewPerInterfaceErrorMap() *PerInterfaceErrorMap {
+	intfErrMap := PerInterfaceErrorMap{}
+	intfErrMap.IntfErrorMap = make(map[string]ErrorInfo)
+	return &intfErrMap
```

Comment:
Where are you using these functions? I see code calling make() directly. And no delete. Having lots of shallow abstraction functions just makes code harder to understand,

Reply:
Everyplace we are creating a PerInterfaceErrorMap.
These are not shallow functions - There are umpteen times where we create a variable - but forget to initialize the map in it.. And then we run into crashes. These kinds of functions help. 

If you look at it - the name itself follows the standard go pattern. Such functions make the code compact and much more readable.